StartSteps-Digital-Education-GmbH/News-App
# Add Endpoint for Commenting on News Articles

**Update:** Implemented the changes after the code review and these are some of the test cases:

<img width="651" alt="Screenshot 2024-12-12 at 11 43 16" src="https://github.com/user-attachments/assets/8c657132-8643-4b55-9e56-a0360eaef204" />

<img width="647" alt="Screenshot 2024-12-12 at 11 43 39" src="https://github.com/user-attachments/assets/dea5572e-c388-495b-aaf6-1e40ee36f327" />

<img width="645" alt="Screenshot 2024-12-12 at 11 43 55" src="https://github.com/user-attachments/assets/5f4c868f-45e5-4c8d-b429-518958b4e359" />

<img width="671" alt="Screenshot 2024-12-12 at 12 02 46" src="https://github.com/user-attachments/assets/49807363-d3d1-4cdf-b2a4-e1ef4afaa1a3" />

**Description:** 

This pull request implements an endpoint for users to comment on news articles. The endpoint allows authenticated users to post a comment on a news article, saving the comment and associated news article in the database. If the news article does not exist in the database, it is created automatically before associating the comment with it.

**Changes:**

1. Added an entity for `Comment` to represent the comment data structure in the database.
2. Created a `route` for comments to handle HTTP requests related to commenting functionality.
3. Developed a `controller` to handle the logic for adding comments on news articles.
4. Added `validation` for the request body using `Joi` to ensure data integrity to required structure.
5. Registered a new main route `(/comments)` in the application to handle all comment-related operations.

**Screenshot:** Below is a screenshot of a successful POST request in Postman, demonstrating the expected behavior of the endpoint:

<img width="717" alt="Screenshot 2024-12-11 at 13 17 14" src="https://github.com/user-attachments/assets/a89254b6-6b14-4721-9eb4-531744357757" />

____

**How to Test:**

*Prerequisites:*

- Register and authenticate a user to obtain a valid token.
- Use Postman or another API client to test the endpoint.
- Include the token in the `Authorization` header: `Bearer <your_token>`.

*Steps:*

Send a `POST` request to the `/comments` endpoint with the following body structure:

```
{
  "title": "News Article Title",
  "description": "Brief description of the article",
  "source": "Article Source",
  "url": "https://example.com/article",
  "publishedAt": "2024-12-10T12:00:00Z",
  "content": "This is my comment"
}
```

*Expected Behavior:*

If the request is valid:

- The news article is created in the database if it does not exist.
- The comment is saved and associated with the news article and the user.
- The response includes the content of the comment and the response message, e.g.:

```
{
  "content": "This is my comment", message: "Comment added successfully"
}
```

*If the request is invalid or the user is not authenticated:*
*Appropriate error messages are returned with corresponding status codes.*

_____

**Please review the implementation and test the functionality. Feedback is welcome!**
